pwiz_tools/Skyline/Controls/GroupComparison/FoldChangeVolcanoPlot.cs

```diff
@@ -593,14 +600,23 @@ protected override void BuildContextMenu(ZedGraphControl sender, ContextMenuStri
                 menuStrip.Items.Insert(index++, new ToolStripMenuItem(GroupComparisonStrings.FoldChangeVolcanoPlot_BuildContextMenu_Formatting___, null, OnFormattingClick));
                 menuStrip.Items.Insert(index++, new ToolStripMenuItem(GroupComparisonStrings.FoldChangeVolcanoPlot_BuildContextMenu_Selection, null, OnSelectionClick)
                     { Checked = Settings.Default.GroupComparisonShowSelection });
+                menuStrip.Items.Insert(index++, new ToolStripMenuItem(GroupComparisonStrings.FoldChangeVolcanoPlot_BuildContextMenu_Avoid_Label_Overlap, null, OnLabelOverlapClick)
+                    { Checked = Settings.Default.GroupComparisonAvoidLabelOverlap });
                 if (AnyCutoffSettingsValid)
                 {
                     menuStrip.Items.Insert(index++, new ToolStripSeparator());
                     menuStrip.Items.Insert(index, new ToolStripMenuItem(GroupComparisonStrings.FoldChangeVolcanoPlot_BuildContextMenu_Remove_Below_Cutoffs, null, OnRemoveBelowCutoffsClick));
-                }  
+                }
+
+
             }
         }
 
+        private void OnLabelOverlapClick(object o, EventArgs eventArgs)
+        {
+            Settings.Default.GroupComparisonAvoidLabelOverlap = !Settings.Default.GroupComparisonAvoidLabelOverlap;
+            UpdateGraph();

```

Comment:
It is possible for the user to have more than one volcano plot showing (if they have more than one Group Comparison) so, technically, you have to update all showing volcano plots when this setting changes.
It might be that the easiest way for all of these volcano plots to update themselves would be to subscribe to the "PropertyChanged" event on "Settings.Default".
Currently, there is no example code in Skyline for listening to property changes on Settings.Default, but I have been thinking for a long time that we should take advantage of that functionality. I usually add listeners like that in the OnHandleCreated event and remove them in the OnHandleDestroyed event.